OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
         }
 
         #endregion
+
+        #region Test GetFeatureVariable<Type> Typecasting
+
+        [Test]
+        public void TestGetFeatureVariableBooleanReturnTypecastedValue()
+        {
```

Comment:
Please add log messages.

Reply:
We can't do that. GetFeatureVariable<Type> methods do not log any message inside them. They just call GetFeatureVariableValueForType and we mocked that method to return the desired value.

Reply:
After discussion with @oakbani, added logging when value casting failed. Log message verification has also been added.